### PR TITLE
[MAINTENANCE] add py311 min versions tests

### PR DESCRIPTION
**Pull request by Kilo59**
- Start testing min versions with python 311
- use xdist for min version tests (for faster test)
- Increase the allowed `timeout` for min version tests (earlier versions of dependencies may not be as performant)
- ~change marker on "slow" `test_onboarding_data_assistant_should_fail_forward` from `unit` to `big`~

### CODE REVIEW

File: tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def test_onboarding_data_assistant_result_batch_id_to_batch_identifier_display_n
     )
 
 
-@pytest.mark.unit
+@pytest.mark.big  # marked as big because this test is regularly taking over 1.5s in CI
```
Comment (Kilo59) [inline tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py]:
```suggestion
@pytest.mark.unit
```

File: ci/constraints-test/py311-min-install.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+numpy==1.25.2
```
Comment (Kilo59) [inline ci/constraints-test/py311-min-install.txt]: Going to decrease this until it starts failing.

Will merge once I find the min version.

Comment (Kilo59) [reply ci/constraints-test/py311-min-install.txt]: `1.23` seems to be the oldest version of `numpy` that works on `3.11`

**Review by Shinnnyshinshin — APPROVED** [review]
🚀

**Pull request merged**